platforms/chibios/drivers/wear_leveling/wear_leveling_efl.c

```diff
@@ -135,7 +143,25 @@ bool backing_store_read(uint32_t address, backing_store_int_t *value) {
     uint32_t             offset = (base_offset + address);
     backing_store_int_t *loc    = (backing_store_int_t *)flashGetOffsetAddress(flash, offset);
     *value                      = ~(*loc);
+
+#ifdef EFL_CHECK_ECC
+    if (ecc_error) {
+        bs_dprintf("Failed to read from backing store, ECC error detected\n");
+        ecc_error = false;
+        return false;
+    }
+#endif // EFL_CHECK_ECC
+
     bs_dprintf("Read  ");
     wl_dump(offset, value, sizeof(backing_store_int_t));
     return true;
 }
+
+void NMI_Handler_EFL(void) {
+#ifdef FLASH_ECCR_ECCD
+    if (FLASH->ECCR & FLASH_ECCR_ECCD) {
+        FLASH->ECCR = FLASH_ECCR_ECCD;
+        ecc_error   = true;
+    }
+#endif // FLASH_ECCR_ECCD
+}
```

Comment:
Maybe we should also have some kind of `ecc_error_expected` flag here, so that any ECC error outside of the backing store range results in an immediate halt instead of attempting to execute random garbage as code?  (It still won't help to distinguish an ECC error in the backing store from an ECC error in the code inside `backing_store_read()` though — for that you would need either some lower level checks, or making `backing_store_read()` run from SRAM.)

Reply:
Yep, done.